File: chapter_2/fibonacciByRecursion.py
def fib(n):
	print(f"fib(n) is called.")
	if n in {0, 1}:
		print(f"Call fib({n}) is returning 1.")
		return 1
	else:
		print(f"Calling fib({n}) and fib({n-1})")
		res = fib(n-1) + fib(n-2)
		print(f"Call to fib({n}) returning ")
		return res

if __name__ == "__main__":
	print(fib(10))

File: chapter_2/fibonacciByRecursion.js
function fib(n) {
	document.write("Called fib(" + n + ")<br />");
	if (n==0 || n == 1) {
		document.write("Returning 1.<br />");
		return 1;
	} else {
		document.write("Calling fib(" + (n-1) + ") " + "and fib(" + (n-2) + ")<br />");
		let res = fib(n-1) + fib(n-2);
		document.write("Call to fib("+ n +") Returning " + res + "<br />");
		return res;
	}
}
document.write(fib(5));
